collector/processor/coldstartprocessor/processor.go

```diff
@@ -0,0 +1,109 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coldstartprocessor // import "github.com/open-telemetry/opentelemetry-lambda/collector/processor/coldstartprocessor"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	semconv "go.opentelemetry.io/collector/semconv/v1.5.0"
+	"go.uber.org/zap"
+)
+
+type faasExecution struct {
+	span     ptrace.Span
+	scope    pcommon.InstrumentationScope
+	resource pcommon.Resource
+}
+
+type coldstartProcessor struct {
+	coldstartSpan *ptrace.Span
+	faasExecution *faasExecution
+	logger        *zap.Logger
+	nextConsumer  consumer.Traces
+}
+
+func (p *coldstartProcessor) processTraces(ctx context.Context, td ptrace.Traces) (ptrace.Traces, error) {
+	var errors error
+	td.ResourceSpans().RemoveIf(func(rs ptrace.ResourceSpans) bool {
```

Comment:
Since this should only ever associate cold start data once per collector instance, would it make sense to add a latch that can be thrown once that work is done and used in a check to short-circuit around this scan of all spans received?

Reply:
Good point, there's now a `bool` flag set once the span has been reported